services/ui-src/src/components/reports/ReportProvider.tsx

```diff
@@ -96,6 +106,56 @@ export const ReportProvider = ({ children }: Props) => {
     }
   };
 
+  const fetchReportForSarCreation = async (selectedState: string) => {
+    try {
+      // clear stored reports by state prior to fetching from current state
+      clearReportsByState();
+
+      const workPlanSubmissions = await getReportsByState(
+        ReportType.WP,
+        selectedState
+      );
+
+      const sarSubmissions = await getReportsByState(
+        ReportType.SAR,
+        selectedState
+      );
+
+      if (workPlanSubmissions?.length !== sarSubmissions?.length) {
+        let lastFoundSubmission: ReportMetadataShape | undefined = undefined;
+        workPlanSubmissions.forEach((submission: ReportMetadataShape) => {
+          if (
+            submission.status === ReportStatus.NOT_STARTED ||
+            submission.status === ReportStatus.IN_PROGRESS
+          ) {
+            if (
+              lastFoundSubmission &&
+              submission.createdAt > lastFoundSubmission?.createdAt
+            )
+              lastFoundSubmission = submission;
+            else if (!lastFoundSubmission) {
+              lastFoundSubmission = submission;
+            }
+          }
+        });
+
+        if (lastFoundSubmission) {
```

Comment:
along with grabbing the most recent WorkPlan with an approved status, we have to make sure that WorkPlan hasn't already been used to create a SAR

Reply:
Good call out. Been thinking maybe add a metadata prop that keeps an id of the associated SAR, and vice-versa. Make for easier data lookup too if we need to.

Reply:
sounds good! We can also address this in a future ticket along with updating the status mentioned above